pkg/component/networking/nodelocaldns/nodelocaldns.go

```diff
@@ -158,368 +190,181 @@ func (n *nodeLocalDNS) Deploy(ctx context.Context) error {
-									},
-									{
-										ContainerPort: int32(53),
-										Name:          "dns-tcp",
-										Protocol:      corev1.ProtocolTCP,
-									},
-									{
-										ContainerPort: int32(prometheusPort),
-										Name:          metricsPortName,
-										Protocol:      corev1.ProtocolTCP,
-									},
-									{
-										ContainerPort: int32(prometheusErrorPort),
-										Name:          errorMetricsPortName,
-										Protocol:      corev1.ProtocolTCP,
-									},
-								},
-								LivenessProbe: &corev1.Probe{
-									ProbeHandler: corev1.ProbeHandler{
-										HTTPGet: &corev1.HTTPGetAction{
-											Host: n.getIPVSAddress(),
-											Path: "/health",
-											Port: intstr.FromInt32(livenessProbePort),
-										},
-									},
-									InitialDelaySeconds: int32(60),
-									TimeoutSeconds:      int32(5),
-								},
-								VolumeMounts: []corev1.VolumeMount{
-									{
-										MountPath: "/run/xtables.lock",
-										Name:      "xtables-lock",
-										ReadOnly:  false,
-									},
-									{
-										MountPath: "/etc/coredns",
-										Name:      "config-volume",
-									},
-									{
-										MountPath: "/etc/kube-dns",
-										Name:      "kube-dns-config",
-									},
-								},
-							},
-						},
-						Volumes: []corev1.Volume{
-							{
-								Name: "xtables-lock",
-								VolumeSource: corev1.VolumeSource{
-									HostPath: &corev1.HostPathVolumeSource{
-										Path: "/run/xtables.lock",
-										Type: &hostPathFileOrCreate,
-									},
-								},
-							},
-							{
-								Name: "kube-dns-config",
-								VolumeSource: corev1.VolumeSource{
-									ConfigMap: &corev1.ConfigMapVolumeSource{
-										LocalObjectReference: corev1.LocalObjectReference{
-											Name: "kube-dns",
-										},
-										Optional: ptr.To(true),
-									},
-								},
-							},
-							{
-								Name: "config-volume",
-								VolumeSource: corev1.VolumeSource{
-									ConfigMap: &corev1.ConfigMapVolumeSource{
-										LocalObjectReference: corev1.LocalObjectReference{
-											Name: configMap.Name,
-										},
-										Items: []corev1.KeyToPath{
-											{
-												Key:  configDataKey,
-												Path: "Corefile.base",
-											},
-										},
-									},
-								},
-							},
-						},
-					},
-				},
-			},
-		}
-		vpa *vpaautoscalingv1.VerticalPodAutoscaler
-	)
+	taskFns = []flow.TaskFn{}
+	// Remove the cleanup-required label from all nodes in this pool in parallel using flow
+	for _, node := range nodeList.Items {
+		node := node // capture range variable
+		taskFns = append(taskFns, func(ctx context.Context) error {
+			patch := client.MergeFrom(node.DeepCopy())
+			delete(node.Labels, labelKeyCleanupRequired)
+			err := n.values.ShootClient.Patch(ctx, &node, patch)
+			if err == nil {
+				n.values.Log.Info("Removed node-local-dns cleanup label from node", "node", node.Name)
+			}
+			return err
```

Comment:
Just return the error, the information should be logged (or follow the `fmt.Errorf` approach that you used before)